server/src/main/java/org/candlepin/async/tasks/HealEntireOrgJob.java

```diff
@@ -0,0 +1,187 @@
+                // of looking up the non or partially compliant products to bind.
+                try {
+                    Consumer consumer = consumerCurator.getConsumer(uuid);
+                    healSingleConsumer(consumer, owner, entitleDate);
+                    result.append("Healing event sent for consumer: ").append(uuid).append("\n");
+                }
+                // We want to catch everything and continue.
+                // Perhaps add something to surface errors later
+                catch (Exception e) {
+                    log.debug("Healing failed for UUID \"{}\" with message: {}", uuid, e.getMessage());
+                    result.append("Healing failed for UUID: ").append(uuid).append("\n");
+                }
+            }
+            return result;
+        }
+        catch (Exception e) {
+            log.error("EntitlerJob encountered a problem.", e);
+            throw new JobExecutionException(e.getMessage(), e, false);
+        }
+    }
+
+    /**
+     * Creates a JobConfig configured to execute the heal entire org job. Callers may further
+     * manipulate the JobConfig as necessary before queuing it.
+     *
+     * @return a JobConfig instance configured to execute the  heal entire org job
+     */
+    public static HealEntireOrgJobConfig createJobConfig() {
+        return new HealEntireOrgJobConfig();
+    }
+
+    /*
+     * Each consumer heal should be a separate transaction
+     */
+    @Transactional
+    private void healSingleConsumer(Consumer consumer, Owner owner, Date date)
+        throws AutobindDisabledForOwnerException {
+        List<Entitlement> ents = entitler.bindByProducts(AutobindData.create(consumer, owner).on(date), true);
+        entitler.sendEvents(ents);
+
+    }
+
+    /**
+     * Job configuration object for the heal entire org job
+     */
+    public static class HealEntireOrgJobConfig extends JobConfig {
+
+        public HealEntireOrgJobConfig() {
+            this.setJobKey(JOB_KEY).setJobName(JOB_NAME)
+                .addConstraint(JobConstraints.uniqueByArguments(OWNER_KEY)); //TODO
+        }
+
+        public HealEntireOrgJobConfig setOwner(final Owner owner) {
+            if (owner == null) {
+                throw new IllegalArgumentException("Owner is null");
+            }
+
+            this.setJobMetadata(LoggingFilter.OWNER_KEY, owner.getKey())
+                .setJobArgument(OWNER_KEY, owner.getKey()).setLogLevel(owner.getLogLevel());
+
+            return this;
+        }
+
+        public HealEntireOrgJobConfig setEntitleDate(final Date entitleDate) {
+            this.setJobArgument(ENTITLE_DATE_KEY, entitleDate);
+
+            return this;
+        }
+
+        @Override
+        public void validate() throws JobConfigValidationException {
+            super.validate();
+
+            try {
+                JobArguments arguments = this.getJobArguments();
+
+                String ownerKey = arguments.getAsString(OWNER_KEY);
+
```

Comment:
In general, we try to not leave too many empty lines between statements, but rather group statements that are logically related together, and separate those groups with empty lines. This is obviously quite subjective, and more like a guidance that is up to you to change if you like, but too many empty lines can hurt readability.

Reply:
Done.